### ISSUE TITLE
Add rule to require `#{}` interpolation for SCSS functions in custom property values

### ISSUE THREAD

**Issue by lyhoang-web-dev** [stylelint/stylelint#9128]
### What is the problem you're trying to solve?

In SCSS, custom property values are treated as opaque strings and are **not processed by the SCSS compiler**. 
This means if a developer writes a SCSS function call inside a custom property value without `#{}` interpolation, it silently outputs the literal function text instead of the evaluated result.

**Example of broken code (no error thrown):**

  ```scss
  // ❌ SCSS does NOT evaluate this — outputs literal "tokens.get(...)"
  --my-color: tokens.get("color.label.fg.secondary");

  Expected behavior:
  // ✅ Correct — SCSS evaluates and interpolates the result
  --my-color: #{tokens.get("color.label.fg.secondary")};
```
This is a silent bug — no compiler error, no warning — yet the output CSS is completely wrong. 
It is easy to miss during code review and hard to debug because the property appears to be set correctly in source.

### What solution would you like to see?

A new rule, e.g. `scss/custom-property-require-interpolation`, that:
  - Targets declarations where the property starts with `--`
  - Detects SCSS function calls in the value (e.g. `tokens.get(`, `color`(,or any `namespace.fn(` pattern)
  - Reports an error if the value is **not** wrapped in `#{...}`

  **Example violations:**
  ```scss
  /* ❌ reported */
  --color: tokens.get("color.fg");
  --size: math.div(16px, 2);

  /* ✅ accepted */
  --color: #{tokens.get("color.fg")};
  --size: #{math.div(16px, 2)};

  /* ✅ accepted — plain CSS value, no SCSS function */
  --color: red;
  --size: 16px;
  ```
  Rule options (optional):
  - `true`: enable with default function detection pattern
  - `{ functionPattern: "regex" }`: allow customizing which function calls
  are flagged

**Comment by jeddy3** [comment]
@lyhoang-web-dev Thanks for the well-written request.

The built-in rules are for standard CSS. The SCSS community maintains a plugin, [`stylelint-scss`](https://github.com/stylelint-scss/stylelint-scss) for SCSS-specific rules. We can't transfer the issue for you as it's a separate GitHub organisation. You'll want to recreate the issue in [their tracker](https://github.com/stylelint-scss/stylelint-scss/issues).
